reconcile/utils/jenkins_api.py

```diff
@@ -33,6 +33,15 @@ def get_job_names(self):
         job_names = [r['name'] for r in res.json()['jobs']]
         return job_names
 
+    def delete_build(self, job_name, build_id):
```

Comment:
do you want to add a retry decorator here?

Reply:
I think in this case I would prefer to let it fail early and we catch it where we use the function. Adding a retry to this would make it inconsistent from the rest of the jenkins_api library

In this specific case it doesn't matter, the next integration run will effectively retry the delete

I personally am not in favor the use of the retry decorator in libraries (vs integration / end-user code). For transient issues they tend to mask the underlying issues by retrying and never bubbling up the issues. We have things that use the retry decorator all over the place and in the end the retry times add up and is part of the reason why things get slow(er) over time and we never know why because we silently retry